subsys/bluetooth/shell/l2cap.c

```diff
@@ -305,15 +372,49 @@ static int cmd_metrics(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_whitelist_add(const struct shell *shell, size_t argc, char *argv[])
+{
+	int i;
+
+	if (!default_conn) {
+		error(shell, "Not connected\n");
+		return 0;
+	}
+
+	for (i = 0; i < ARRAY_SIZE(l2cap_whitelist); i++) {
+		if (l2cap_whitelist[i] == NULL) {
+			l2cap_whitelist[i] = bt_conn_ref(default_conn);
+			return 0;
+		}
+	}
+
+	return -ENOMEM;
+}
+
+static int cmd_whitelist_remove(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!default_conn) {
+		error(shell, "Not connected\n");
+		return 0;
+	}
+
+	l2cap_whitelist_remove(default_conn, 0);
+
+	return 0;
+}
+
 #define HELP_NONE "[none]"
 
 SHELL_CREATE_STATIC_SUBCMD_SET(l2cap_cmds) {
 	SHELL_CMD(connect, NULL, "<psm>", cmd_connect),
 	SHELL_CMD(disconnect, NULL, HELP_NONE, cmd_disconnect),
 	SHELL_CMD(metrics, NULL, "<value on, off>", cmd_metrics),
 	SHELL_CMD(recv, NULL, "[delay (in miliseconds)", cmd_recv),
-	SHELL_CMD(register, NULL, "<psm> [sec_level]", cmd_register),
+	SHELL_CMD(register, NULL, "<psm> [sec_level] "
+		  "[policy: whitelist, 16byte_key]", cmd_register),
 	SHELL_CMD(send, NULL, "<number of packets>", cmd_send),
+	SHELL_CMD(whitelist-add, NULL, HELP_NONE, cmd_whitelist_add),
+	SHELL_CMD(whitelist-remove, NULL, HELP_NONE, cmd_whitelist_remove),
```

Comment:
I would create `whitelist` command with `add` and `remove` subcommands :) Bu this is only my approach.

Reply:
nice :) done